src/jdk.jpackage/share/native/applauncher/ReleaseFile.cpp

```diff
@@ -0,0 +1,174 @@
+
+        tstring_array rvers = tstrings::split(reqVer, _T("."));
+        tstring_array overs = tstrings::split(version, _T("."));
+
+        tstring_array::const_iterator rit = rvers.begin();
+        const tstring_array::const_iterator rend = rvers.end();
+
+        tstring_array::const_iterator oit = overs.begin();
+        const tstring_array::const_iterator oend = overs.end();
+
+        for (; rit != rend; ++rit) {
+            int rnum = stoi(*rit);
+            const tstring oitstr = (oit != oend) ? *oit++ : _T("0");
+            int onum = stoi(oitstr);
+            if (greaterThan) {
+                if (rnum > onum) {
+                    return false;
+                }
+            } else {
+                if (rnum != onum) {
+                    return false;
+                }
+            }
+        }
+    }
+    return true;
+}
+
+bool ReleaseFile::greaterThan(const tstring& v1, const tstring& v2) {
+    tstring_array vers1 = tstrings::split(v1, _T("."));
+    tstring_array vers2 = tstrings::split(v2, _T("."));
+
+    tstring_array::const_iterator it1 = vers1.begin();
+    const tstring_array::const_iterator end1 = vers1.end();
+
+    tstring_array::const_iterator it2 = vers2.begin();
+    const tstring_array::const_iterator end2 = vers2.end();
+
+    for (; it1 != end1; ++it1) {
+        int num1 = stoi(*it1);
+        const tstring num2str = (it2 != end2) ? *it2++ : _T("0");
+        int num2 = stoi(num2str);
+        if (num1 > num2) {
+            return true;
+        }
+        if (num1 < num2) {
+            return false;
+        }
+    }
+    return false;
+}
+
+bool ReleaseFile::satisfies(ReleaseFile required, const tstring& versionSpec) {
+    // We need to insure version satisfies the versionSpec
+    if (versionMatch(versionSpec, getVersion())) {
+        // now we need to make sure all required modules are there.
+        tstring_array reqmods = required.modules;
+        tstring_array canmods = modules;
+
+        for (int i=0; i<(int)reqmods.size(); i++) {
+            int j = 0;
+            for (; j<(int)canmods.size(); j++) {
+                if (tstrings::equals(reqmods[i], canmods[j])) {
+                    break;
+                }
+            }
+            if (j == (int)canmods.size()) {
+                LOG_TRACE(tstrings::any() << " missing mod: " << reqmods[i]
+                        << "in version: " << getVersion());
+                return false;
+            }
+        }
+        LOG_TRACE(tstrings::any() << " all modules satisfied with: "
+                                  << getVersion());
+        return true;
+    }
+    return false;
+}
+
+ReleaseFile ReleaseFile::load(const tstring& path) {
+    std::ifstream input(path.c_str());
+
+    ReleaseFile releaseFile;
+
+    if (input.good()) {
+        std::string utf8line;
+
+        int lineno = 0;
+        // we should find JAVA_VERSION and MODULES in the first few lines
+        while ((std::getline(input, utf8line)) && lineno++ < 10) {
```

Comment:
I'd move `10` to enum to make this limit explicit.

Reply:
yes